docs/helpers.md

```diff
@@ -6,72 +6,106 @@ docsify extends Markdown syntax to make your documents more readable.
 
 ## Callouts
 
-### Important content
+Docsify supports [GitHub style](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts) callouts (also known as "admonitions" or "alerts").
 
-Important content like:
+<!-- prettier-ignore -->
+> [!CAUTION]
+> **Caution** callouts communicate negative potential consequences of an action.
 
-```markdown
-!> **Time** is money, my friend!
-```
+<!-- prettier-ignore -->
+> [!IMPORTANT]
+> **Important** callouts communicate information necessary for users to succeed.
 
-is rendered as:
+<!-- prettier-ignore -->
+> [!NOTE]
+> **Note** callouts communicate information that users should take into account.
 
-!> **Time** is money, my friend!
+<!-- prettier-ignore -->
+> [!TIP]
+> **Tip** callouts communicate optional information to help a user be more successful.
 
-### Tips
+<!-- prettier-ignore -->
+> [!WARNING]
+> **Warning** callouts communicate potential risks user should be aware of.
 
-General tips like:
+**Markdown**
 
+<!-- prettier-ignore -->
 ```markdown
-?> _TODO_ unit test
+> [!CAUTION]
+> **Caution** callouts communicate negative potential consequences of an action.
+
+> [!IMPORTANT]
+> **Important** callouts communicate information necessary for users to succeed.
+
+> [!NOTE]
+> **Note** callouts communicate information that users should take into account.
+
+> [!TIP]
+> **Tip** callouts communicate optional information to help a user be more successful.
+
+> [!WARNING]
+> **Warning** callouts communicate potential risks user should be aware of.
 ```
 
-are rendered as:
+### Legacy Style ⚠️
 
-?> _TODO_ unit test
+The following Docsify v4 callout syntax has been deprecated and will be removed in a future version.
```

Comment:
I think it would be great to add a `console.warn` when the respective piece of code runs, so people will know. We could also tell people to use major versions in deprecation messages, to start to get versionless people knowing what will be happening, so they can prepare. Wdyt?

Reply:
> I think it would be great to add a console.warn when the respective piece of code runs

Easy enough. Happy to add it. I may consolidate console deprecation warnings as well so they are easier to track and the output can be standardized.

> We could also tell people to use major versions in deprecation messages, to start to get versionless people knowing what will be happening

This one may introduce some false positives simply because of the many different ways Docsify can be loaded, but I think a simple search for a `<script>` tag with a `src` value that matches the following criteria could work:

- Source domain is different than route domain (i.e., markdown page domain)
- Source domain contains a TLD (e.g. `.com`, `.net`, `.org`, etc.)
- Source value matches `\\docsify(\.min)?\.js`
- Source match is not preceded by a number (covers `@version/` and `/version/` CDN URLs)

This would match the following URLs and trigger the console warning:

- https://static.domain.com/docsify.min.js 👈 Could be false positive
- https://cdn.jsdelivr.net/npm/docsify/docsify.min.js
- https://unpkg.com/docsify/docsify.min.js
- https://cdn.bootcdn.net/ajax/libs/docsify/docsify.min.js
- https://cdnjs.cloudflare.com/ajax/libs/docsify/docsify.min.js

The following URLs would be ignored:

- https://localhost/docsify.min.js
- https://10.10.10.1/docsify.min.js
- https://cdn.jsdelivr.net/npm/docsify@5/docsify.min.js
- https://unpkg.com/browse/docsify@5.0.0/docsify.min.js
- https://cdn.bootcdn.net/ajax/libs/5/docsify/docsify.min.js
- https://cdnjs.cloudflare.com/ajax/5.0.0/libs/docsify/docsify.min.js

FWIW, I may opt to implement this in a separate PR. Replying here for convenience. :)

Reply:
> We could also tell people to use major versions in deprecation messages, to start to get versionless people knowing what will be happening

> This one may introduce some false positives simply because of the many different ways Docsify can be loaded, but I think a simple search for a `<script>` tag with a `src` value that matches the following criteria could work...

I think it seems more complicated, or am I too straightforward?
Firstly, We check some common CDN (unpack, jsdelivr...etc) import path format with path folder version.
For other importing, we check the `docsify.min.js` and the `docsify.js` content, no matter what the path folder `/5/` nor the domain is, cos we don't know users how to host docsify from which server or which place, which Ip, which domin, which DNS, which CDN...



When the site contains either resources name of them, we throw out a warning such as:
```
[docsifty] versionless importing detect:
Use versionless make get the unpredictable breaking changes, please use the major versions or strict to one version instead.
More details: https://docsifyjs.org/docs/quickstart.md#specifying-docsify-versions
```
Result: 
If they import the resource online (host site on a domain or a raw IP are both okay ), they would get the warning. 
If they deploy docsify internally or locally, they won't get a docsify warning, cos they won't get any docsify update also, they versioning is doesn't matter for them. Once they manually upgrade docsify, they would get the warning then.

--- 
And we provide a `versionDetector` config to manually suppress the warning if they dont care about any breaking changes or the resource import place is not a generic resources versioning control path (e.g. https://dontblameme.com/libs/docsify@4.min.js).
```js
window.$docsify = {
  versionDetector: false
};

```